bqskit/qis/pauliz.py

```diff
@@ -0,0 +1,251 @@
+"""This module implements the PauliMatrices class."""
+from __future__ import annotations
+
+import itertools as it
+from typing import Iterable
+from typing import Iterator
+from typing import overload
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import numpy as np
+import numpy.typing as npt
+
+from bqskit.utils.typing import is_integer
+from bqskit.utils.typing import is_numeric
+from bqskit.utils.typing import is_sequence
+
+if TYPE_CHECKING:
+    from bqskit.qis.unitary.unitary import RealVector
+
+
+class PauliZMatrices(Sequence[npt.NDArray[np.complex128]]):
+    """
+    The group of Pauli Z matrices.
+
+    A PauliZMatrices object represents the entire of set of Pauli Z matrices for
+    some number of qubits.
```

Comment:
This is not immediately clear what this is. Maybe give an example for 1, and say 3 qubits?

Additionally, if there are relevant references, list them here.